app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PicturesController < ApplicationController
+  before_filter :authenticate_user!, except: [:index, :show]
+  before_action :set_picture, only: [:show, :edit, :update, :destroy]
+
+  # GET /pictures
+  # GET /pictures.json
+  def index
+    @pictures = Picture.all
+  end
+
+  # GET /pictures/1
+  # GET /pictures/1.json
+  def show
+  end
+
+  # GET /pictures/new
+  def new
+    @picture = Picture.new
+  end
+
+  # GET /pictures/1/edit
+  def edit
+  end
+
+  # POST /pictures
+  # POST /pictures.json
+  def create
+    @picture = Picture.new(picture_params)
+    # binding.pry
+    respond_to do |format|
+      if @picture.save
+        format.html { redirect_to @picture, notice: 'Picture was successfully created.' }
```

Comment:
Metrics/LineLength: Line is too long. [89/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)